{{cookiecutter.project_slug}}/Dockerfile

```diff
@@ -1,136 +1,142 @@
       pip \
       wheel \
       setuptools
 
-WORKDIR /home/scu
+WORKDIR /build
 
-# install base 3rd party packages to accelerate runtime installs
-COPY --chown=scu:scu services/{{ cookiecutter.project_slug }}/docker docker
-COPY --chown=scu:scu services/{{ cookiecutter.project_slug }}/requirements/base.txt requirements-base.txt
+# install base 3rd party dependencies
+COPY  services/{{ cookiecutter.project_slug }}/requirements/*.txt \
+      services/{{ cookiecutter.project_slug }}/requirements/
 
-RUN $PIP install -r requirements-base.txt &&\
-    rm -rf /tmp/.pipcache
+RUN $SC_PIP install \
+      -r services/{{ cookiecutter.project_slug }}/requirements/base.txt
 
 
 # --------------------------Cache stage -------------------
 # CI in master buils & pushes this target to speed-up image build
 #
+#  + /build
+#    + services/{{ cookiecutter.project_slug }} [scu:scu] WORKDIR
+#
 FROM build as cache
 
-COPY --chown=scu:scu packages $HOME/packages
-COPY --chown=scu:scu services/{{ cookiecutter.project_slug }} $HOME/services/{{ cookiecutter.project_slug }}
+ENV SC_BUILD_TARGET cache
 
-WORKDIR /home/scu/services/{{ cookiecutter.project_slug }}
+{# TODO: Consider .dockerignore filter to avoid #}
+COPY --chown=scu:scu packages /build/packages
+COPY --chown=scu:scu services/{{ cookiecutter.project_slug }} /build/services/{{ cookiecutter.project_slug }}
 
-RUN $PIP install -r requirements/base.txt &&\
-    $PIP install . &&\
-    $PIP list &&\
-    rm -rf /tmp/.pipcache
+WORKDIR /build/services/{{ cookiecutter.project_slug }}
 
-WORKDIR /home/scu
-ENV BOOT_MODE production
+{# NOTE: notice that tests are not installed within the docker. Currently pytest runs in the host #}
+RUN $SC_PIP install . &&\
+    $SC_PIP list -v
 
 
 # --------------------------Production stage -------------------
 # Final cleanup up to reduce image size and startup setup
-# TODO: optimize size by uninstalling dev tools? https://wiki.alpinelinux.org/wiki/Alpine_Linux_package_management#Remove_a_Package
+# Runs as scu (non-root user)
+#
+#  + /home/scu     $HOME = WORKDIR
+#    + services/{{ cookiecutter.project_slug }} [scu:scu]
 #
 FROM cache as production
+{# TODO: optimize size by uninstalling dev tools? https://wiki.alpinelinux.org/wiki/Alpine_Linux_package_management#Remove_a_Package #}
+
+ENV SC_BUILD_TARGET production
+ENV SC_BOOT_MODE production
+
+WORKDIR /home/scu
+
+RUN mkdir -p services/{{ cookiecutter.project_slug }} &&\
+    chown scu:scu services/{{ cookiecutter.project_slug }} &&\
+    mv /build/services/{{ cookiecutter.project_slug }}/docker services/{{ cookiecutter.project_slug }}/docker &&\
+    rm -rf /build
 
 RUN apk del --no-cache\
       gcc \
       git
 
-RUN $PIP list &&\
-    rm -rf $HOME/packages &&\
-    rm -rf $HOME/services/{{ cookiecutter.project_slug }} &&\
-    rm -rf /tmp/.pipcache
 
-ENTRYPOINT [ "/bin/sh", "docker/entrypoint.sh" ]
-CMD ["/bin/sh", "docker/boot.sh"]
+HEALTHCHECK --interval=30s \
+            --timeout=120s \
```

Comment:
120s is a very long timeout no? I would default to something like 10-20s instead. I mean the detection of unhealthiness should be fast